modules/dxp/apps/search-experiences/search-experiences-rest-impl/src/main/java/com/liferay/search/experiences/rest/internal/resource/v1_0/FieldMappingInfoResourceImpl.java

```diff
@@ -51,18 +52,27 @@
 	extends BaseFieldMappingInfoResourceImpl {
 
 	@Override
-	public Page<FieldMappingInfo> getFieldMappingInfosPage(String query)
+	public Page<FieldMappingInfo> getFieldMappingInfosPage(
+			Boolean external, String indexName, String query)
 		throws Exception {
 
-		return Page.of(getFieldMappings(query));
+		return Page.of(getFieldMappings(external, indexName, query));
 	}
 
-	public List<FieldMappingInfo> getFieldMappings(String query) {
+	public List<FieldMappingInfo> getFieldMappings(
+		boolean external, String indexName, String query) {
+
 		JSONObject jsonObject = FieldMappingsWebCacheItem.get(
-			_indexInformation,
-			_indexNameBuilder.getIndexName(contextCompany.getCompanyId()),
+			_indexInformation, _getIndexName(external, indexName),
 			_jsonFactory);
 
+		if (jsonObject == null) {
+			jsonObject = FieldMappingsWebCacheItem.get(
+				_indexInformation,
+				_indexNameBuilder.getIndexName(contextCompany.getCompanyId()),
+				_jsonFactory);
+		}
```

Comment:
@wcao20170619 not sure if this is necessary. If _getIndexName() works and returns a valid index name, then i don't see a need for a fallback. Also, FieldMappingsWebCacheItem.get() should not need null check because it always return a JSONObject.

Reply:
@peerkar is it possible that an external index name is in a Blueprint's configuration but it does not exist in storage? If it is not the case. I can remove it.